sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -746,6 +750,337 @@ case class DescribeTableCommand(
+        .map(defaultValue => s""", "default_value": "${defaultValue.replace("\"", "\\\"")}"""")
+        .getOrElse("")
+
+      s"""{
+         |  "id": ${id + 1},
+         |  "name": "${column.name}",
+         |  "type": ${jsonType(column.dataType)}
+         |  $commentField$defaultValueJson
+         |}""".stripMargin
+    }.mkString("[", ",", "]")
+
+    addKeyValueToJson(buffer, "columns", parse(columnsJson))
+  }
+
+  protected def appendJson(buffer: ArrayBuffer[Row], key: String, jsonObject: String): Unit = {
+    val currentJson = buffer.headOption.map(row => parse(row.getString(0))).
+      getOrElse(parse("""{}"""))
+    val newJson = parse(jsonObject.replace(" ", ""))
+    val updatedJson = currentJson merge JObject(key -> newJson)
+
+    if (buffer.isEmpty) {
+      buffer += Row(compact(render(updatedJson)))
+    } else {
+      buffer(0) = Row(compact(render(updatedJson)))
+    }
+  }
+
+  private def addKeyValueToJson(buffer: ArrayBuffer[Row], key: String, value: JValue): Unit = {
+    val normalizedKey = normalizeStr(key)
+
+    val currentJson = buffer.headOption.map(row => parse(row.getString(0))).getOrElse(JObject())
+    if ((currentJson \ normalizedKey) == JNothing) {
+      val updatedJson = currentJson merge JObject(normalizedKey -> value)
+      val jsonString = compact(render(updatedJson))
+
+      if (buffer.isEmpty) {
+        buffer += Row(jsonString)
+      } else {
+        buffer(0) = Row(jsonString)
+      }
+    }
+  }
+
+  private def describeClusteringInfoJson(table: CatalogTable, buffer: ArrayBuffer[Row]): Unit = {
+    table.clusterBySpec.foreach { clusterBySpec =>
+      val clusteringColumnsJson = clusterBySpec.columnNames.map { fieldNames =>
+        val nestedFieldOpt = table.schema.findNestedField(fieldNames.fieldNames.toIndexedSeq)
+        assert(nestedFieldOpt.isDefined,
+          "The clustering column " +
+            s"${fieldNames.fieldNames.map(quoteIfNeeded).mkString(".")} " +
+            s"was not found in the table schema ${table.schema.catalogString}."
+        )
+        val (path, field) = nestedFieldOpt.get
+        s"""{
+           |  "name": "${(path :+ field.name).map(quoteIfNeeded).mkString(".")}",
+           |  "type": ${jsonType(field.dataType)},
+           |  "comment": ${field.getComment().map(c => s""""$c"""").getOrElse("null")}
+           |}""".stripMargin
+      }.mkString("[", ",", "]")
+
+      appendJson(buffer, "clustering_information", clusteringColumnsJson)
+    }
+  }
+
+  private def describeFormattedTableInfoJson(
+        table: CatalogTable, buffer: ArrayBuffer[Row]): Unit = {
+
+    val excludedTableInfo = Set("catalog", "schema", "database", "table", "location",
+      "serde_library", "inputformat", "outputformat")
+
+    table.bucketSpec match {
+      case Some(spec) =>
+        spec.toJsonLinkedHashMap.map { case (key, value) =>
+          val jsonValue: JValue =
+            Try(parse(value)) match {
```

Comment:
I don't quite get the idea here. So `toJsonLinkedHashMap` returns JSON string and we parse it here, but why not just call `toLinkedHashMap` and simply wrap the string values with `JString` so that we don't need to parse it?

Reply:
`toLinkedHashMap` doesn't parse dict strings into a json compatible format, e.g. for this existing describe table output we need to do our own parsing into dictionary format for JSON:
```
Table Properties    	[e=3, password=*********(redacted), t=test]
```

Also, for the below to be valid, it should be a JSON array of strings, so we should wrap the column names in "..." rather than `...`:
```
Sort Columns        	[`b`]  
```
Also `value` can be a string / array / dict, so we'd need to handle cases (JString, JArray, JObject etc.) for this (may as well abstract it to the json4s `parse()` )

Reply:
Ah I see. I think we should have a more general `CatalogTable#toJsonLinkedHashMap` that returns `Map[String, JValue]`, and then `toLinkedHashMap` just calls `toJsonLinkedHashMap` and turn `JValue` into String as it likes.